extensions/binding/org.eclipse.smarthome.binding.astro.test/src/test/groovy/org/eclipse/smarthome/binding/astro/test/AstroStateTest.groovy

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.astro.test
+
+import static org.eclipse.smarthome.binding.astro.test.cases.AstroParametrizedTestCases.*
+import static org.hamcrest.CoreMatchers.*
+import static org.junit.Assert.*
+
+import java.time.LocalDateTime
+import java.time.ZonedDateTime
+
+import org.eclipse.smarthome.binding.astro.AstroBindingConstants
+import org.eclipse.smarthome.binding.astro.internal.calc.MoonCalc
+import org.eclipse.smarthome.binding.astro.internal.calc.SunCalc
+import org.eclipse.smarthome.binding.astro.internal.config.AstroChannelConfig
+import org.eclipse.smarthome.binding.astro.internal.model.Planet
+import org.eclipse.smarthome.binding.astro.internal.util.PropertyUtils
+import org.eclipse.smarthome.binding.astro.test.cases.AstroParametrizedTestCases
+import org.eclipse.smarthome.core.thing.ChannelUID
+import org.eclipse.smarthome.core.thing.ThingUID
+import org.eclipse.smarthome.core.types.State
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import org.junit.runners.Parameterized.Parameters
+
+/**
+ * Tests for the Astro Channels state
+ *
+ * @See {@link AstroParametrizedTestCases}
+ * @author Petar Valchev - Initial Implementation
+ * @author Svilen Valkanov - Added missing configuration parameters and replaced headers
+ *
+ */
+@RunWith(Parameterized.class)
+class AstroStateTest {
```

Comment:
I agree that the name could be better, but I think `PropertyUtilsTest` may be misleading, as calculations are involved as well. (that are not part of the PropertyUtils).